spec/lib/miq_ldap_spec.rb

```diff
@@ -119,4 +119,20 @@
     data = "\001\005\000\000\000\000\000\005\025\000\000\000+\206\301\364y\307\r\302=\336p\216\237\004\000\000"
     expect(MiqLdap.sid_to_s(data)).to eq("S-1-5-21-4106323499-3255682937-2389761597-1183")
   end
+
+  it 'returns a hostname when a hostname is availble and does not disable verify mode' do
+    allow(TCPSocket).to receive(:gethostbyname).and_return(["testhostname", "aliases", "type", "192.168.252.20"])
+    allow(TCPSocket).to receive(:new)
+    ldap = MiqLdap.new(:host => ["testhostname", "localhost", "dummy", @host])
+    expect(ldap.ldap.host).to eq("testhostname")
+    expect(ldap.ldap.instance_variable_get(:@encryption).try(:has_key_path?, :tls_options, :verify_mode)).to be_falsey
+  end
+
+  it 'returns an IPAddress and disables verify mode when only an IPAddress is availble' do
+    expect(TCPSocket).not_to receive(:gethostbyname)
+    allow(TCPSocket).to receive(:new)
+    ldap = MiqLdap.new(:host => ["192.168.254.15", "localhost", "dummy", @host])
+    expect(ldap.ldap.host).to eq("192.168.254.15")
+    expect(ldap.ldap.instance_variable_get(:@encryption).fetch_path(:tls_options, :verify_mode)).to eq(0)
```

Comment:
can eq(0) be eq(OpenSSL::SSL::VERIFY_NONE)  ?  it will be clearer.

Reply:
👍